src/components/Snippet/FileManager.vue

```diff
@@ -0,0 +1,215 @@
+<script>
+import { setTimeout } from 'timers';
+
+import {
+  Upload as GirderUpload,
+  UpsertFolder as GirderUpsertFolder,
+  DataBrowser as GirderDataBrowser,
+  Breadcrumb as GirderBreadcrumb,
+} from '../';
+import {
+  getLocationType,
+  isRootLocation,
+  createLocationValidator,
+} from '../../utils';
+
+export default {
+  components: {
+    GirderBreadcrumb,
+    GirderUpload,
+    GirderUpsertFolder,
+    GirderDataBrowser,
+  },
+
+  props: {
+    location: {
+      type: Object,
+      validator: createLocationValidator(true),
+      default: () => { 'root'; },
+    },
+    rootLocationAllowed: {
```

Comment:
> it could have an invalid initial state. 

Not sure I understand.   The created hook check (that I need to add) will throw if location is root and this prop is false.  What is the consequence of making this default to false?

If it defaults to true and explicitly set it as false but pass in a root location, don't you end up in the same invalid state?

Reply:
I meant that because rootLocationAllowed is an optional prop if the consumer doesn't set it and don't set the location, the location will default to root, and the allowed will be false, making the FileManager state invalid at the start. A component should not be invalid if an optional prop is not set. 

> If it defaults to true and explicitly set it as false but pass in a root location, don't you end up in the same invalid state?
Good question, in that case, it is the consumer who disabled the root and passes a root location (or don't pass since it's the default), so we could use one of the validators to throw an error, and the consumer should take the blame because it's their own action.

Reply:
Ok, I agree with that.  Will change `rootLocationAllowed` to `rootLocationDisabled` universally.

Reply:
Not necessary universal, because `DataBrowser` and `Breadcrumb` won't have this `initial invalid state` issue, change them universally if you would like, otherwise, just change this one is ok to me.